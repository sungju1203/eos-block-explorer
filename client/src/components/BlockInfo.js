import React from "react"
import { Query } from "react-apollo"
import gql from "graphql-tag"

const GET_BLOCK_INFO = gql`
  {
    block {
      block_num
      timestamp
      producer
    } 
  }
`

const BlockInfo = () => (
  <Query query={GET_BLOCK_INFO}>
    {({ loading, error, data }) => {
      console.log("L: ", loading)
      console.error("E: ", error)
      console.info("Data: ", data)
      if (loading) return <p>Loading...</p>
      if (error) return <p>Error :(</p>
      return <div> {data.block.block_num} / {data.block.timestamp} / {data.block.producer} </div>
      // return data.rates.map(({ id, block_num }) => (
      //   <div key={currency}>
      //     <p>{`${id}: ${block_num}`}</p>
      //   </div>
      // ))
    }}
  </Query>
)

export default BlockInfo