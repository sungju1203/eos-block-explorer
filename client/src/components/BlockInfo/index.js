import React, { Component } from "react"
import { Query } from "react-apollo"
import gql from "graphql-tag"
import styled from "styled-components"
import moment from "moment"

import BlockNumber from "./BlockNumber"
import BlockDetail from "./BlockDetail"
import { Button, H3 } from "theme/globalStyle"
import blockIcon from "images/block-icon.png"

const GET_BLOCK_INFO = gql`
  {
    blocks {
      block_num
      timestamp
      producer
    }
  }
`

const Wrapper = styled.div`
  width: 80vw;
`

const Header = styled.div`
  background: ${props => props.theme.white};
  padding: 5px;
  display: flex;
  justify-content: space-between;
  align-items: center;
`

const IconWrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`

const BlockIcon = styled.img`
  width: 30px;
  height: 30px;
  margin-right: 0.5rem;
`

const InfoWrapper = styled.div`
  background: ${props => props.theme.white};
  display: flex;
  margin: 0.5rem 0;
  &:hover {
    cursor: pointer;
  }
`

class BlockContainer extends Component {
  state = { isLoaded: false }

  fetchBlockInfo = () => {
    this.setState({ isLoaded: true })
  }

  showDetail = async () => {
    
  }

  render() {
    const { isLoaded } = this.state

    return (
      <Query query={GET_BLOCK_INFO}>
        {({ error, data, refetch }) => {
          if (error) return <p>Fetching Error</p>
          return (
            <Wrapper>
              <Header>
                <IconWrapper>
                  <BlockIcon src={blockIcon} />
                  <H3>Blocks</H3>
                </IconWrapper>
                {isLoaded ?
                  <Button onClick={() => refetch()}>RELOAD</Button> :
                  <Button onClick={this.fetchBlockInfo}>LOAD</Button>
                }
              </Header>
              {isLoaded &&
                data.blocks.map(({ block_num, timestamp, producer }) => (
                  <InfoWrapper key={block_num} onClick={() => this.showDetail()} >
                    <BlockNumber blockNumber={block_num} />
                    <BlockDetail timestamp={moment(timestamp).format("LLL")} producer={producer} />
                  </InfoWrapper>
                ))
              }
            </Wrapper>
          )
        }}
      </Query>
    )
  }
}

export default BlockContainer