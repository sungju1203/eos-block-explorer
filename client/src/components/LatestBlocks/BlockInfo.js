import React from "react"
import styled from "styled-components"
import PropTypes from "prop-types"

import { InfoText } from "theme/globalStyle"

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding-left: 1rem;
`

const TextWrapper = styled.div`
  display: flex;
`

const BlockInfo = ({ id, timestamp, producer, transactions }) => (
  <Wrapper>
    <TextWrapper>
      produced by &nbsp;
      <InfoText>{producer}</InfoText>
    </TextWrapper>
    <TextWrapper>
      at &nbsp;
      <InfoText>{timestamp}</InfoText>
    </TextWrapper>
    <TextWrapper>
      <InfoText>{transactions}</InfoText>
       &nbsp; transactions
    </TextWrapper>
  </Wrapper>
)

BlockInfo.propTypes = {
  timestamp: PropTypes.string.isRequired,
  producer: PropTypes.string.isRequired,
  transactions: PropTypes.number.isRequired,
}

export default BlockInfo