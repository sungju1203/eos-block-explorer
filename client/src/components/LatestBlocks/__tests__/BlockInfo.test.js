import React from "react"
import renderer from "react-test-renderer"

import BlockInfo from "../BlockInfo"


test('Block Info', () => {
  const props = {
    id: '0015ef887c1badff4d207fc03af2ceb78f935ba380e45b71b601901ddb7e508d',
    timestamp: '2018-05-17T04:34:36.500',
    producer: 'eosio',
    transactions: 0,
  }

  const component = renderer.create(
    <BlockInfo {...props} />,
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})