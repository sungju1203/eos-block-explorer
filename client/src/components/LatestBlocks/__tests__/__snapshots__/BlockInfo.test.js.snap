// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Block Info 1`] = `
<div
  className="sc-bxivhb czdAHg"
>
  <div
    className="sc-ifAKCX WOpjk"
  >
    produced by  
    <div
      className="sc-htpNat iELaDV"
    >
      eosio
    </div>
  </div>
  <div
    className="sc-ifAKCX WOpjk"
  >
    at  
    <div
      className="sc-htpNat iELaDV"
    >
      2018-05-17T04:34:36.500
    </div>
  </div>
  <div
    className="sc-ifAKCX WOpjk"
  >
    <div
      className="sc-htpNat iELaDV"
    >
      0
    </div>
      transactions
  </div>
</div>
`;
